pennylane/tape/qscript.py

```diff
@@ -1331,6 +1334,21 @@ def from_queue(cls, queue):
         return cls(*process_queue(queue))
 
 
+class SpecsDict(dict):
+    """A dictionary to track and warn about deprecated keys"""
+
+    deprecated_keys = ("gate_types", "num_operations", "num_used_wires", "depth")
+    new_keys = ("gate_types", "num_gates", "num_wires", "depth")
+
+    def __getitem__(self, item):
+        if item in self.deprecated_keys:
+            warnings.warn(
+                f"The {item} key is deprecated and will be removed in the next release. "
```

Comment:
Which release? I think this should state the number.

Reply:
It will be removed in V0.32! I don't know if this is a standard practice? I feel like it might be safer to leave it out incase the deprecation gets postponed?

Reply:
Mh. Good point :sweat_smile: 
Looking through the codebase, you seem to be right and we don't give version numbers usually :)